pkg/platform/common/vmlayer/cloudlet.go

```diff
@@ -89,9 +89,28 @@ func (v *VMPlatform) GetPlatformNodes(cloudlet *edgeproto.Cloudlet) []NodeInfo {
 	return nodes
 }
 
+// GetCloudletImageName decides what image to use based on
+// 1) if MEX_OS_IMAGE is specified in properties and not default, use that
+// 2) Use image specified on startup based on cloudlet config
+func (v *VMPlatform) GetCloudletImageName(ctx context.Context) (string, error) {
+	imgFromProps := v.VMProperties.GetCloudletOSImage()
+	if imgFromProps != "" {
+		log.SpanLog(ctx, log.DebugLevelInfra, "using image from MEX_OS_IMAGE property", "imgFromProps", imgFromProps)
+		return imgFromProps, nil
+	}
+
+	vmBaseImage := v.VMProperties.CommonPf.PlatformConfig.CloudletVMImagePath
+	if vmBaseImage == "" {
+		return "", fmt.Errorf("Get cloudlet image failed, cloudletVMImagePath not set")
+	}
+	imageNameWithoutExt := util.RemoveExtension(filepath.Base(vmBaseImage))
+	return imageNameWithoutExt, nil
```

Comment:
this code is duplicated on lines 115-124, I suggest you remove those lines and replace them with this new function.

Reply:
So I was thinking about this, but the logic becomes more complicated in `GetCloudletImageToUse`. But adding an extra return value made the code ok-ish.